pkg/volume/awsebs/aws_ebs_test.go

```diff
@@ -376,3 +378,38 @@ func TestMountOptions(t *testing.T) {
 		t.Errorf("Expected mount options to be %v got %v", expectedMountOptions, mountOptions)
 	}
 }
+
+func TestGetCandidateZone(t *testing.T) {
+	const testZone = "my-zone-1a"
+	const testPVCName = "my-pvc"
+
+	// TODO: add test case for immediate bind volume when we have a good way to mock Cloud outside cloudprovider
+	tests := []struct {
+		cloud         *aws.Cloud
+		node          *v1.Node
+		expectedZones sets.String
+	}{
+		{
+			cloud: nil,
+			node: &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						v1.LabelZoneFailureDomain: testZone,
+					},
+				},
+			},
+			expectedZones: sets.NewString(),
```

Comment:
The tests don't seem to be useful to me if they check the zones to be empty.

Reply:
so the reasoning here for the function is - if we have a selected node, there should not be a "candidate zone" returned, and if not (bind immediately), we should compute candidate zone. once we have better way to mock aws cloud provider, we need to add other test cases.

Or since the logic is rather simple, I can remove the added helper and combine the logic in `CreateVolume` so we can remove the test. What do you think?